service/src/main/java/com/ge/predix/acs/privilege/management/dao/AttributeTraversalException.java

```diff
@@ -0,0 +1,14 @@
+package com.ge.predix.acs.privilege.management.dao;
+
+public class AttributeTraversalException extends RuntimeException {
```

Comment:
Semantically speaking, we are traversing a graph - not attributes. Maybe GraphTraversalException would make more sense? Graphs are well-known data structures so users might be able to understand graph traversal, a well-known problem in CS, than attribute traversal (something we more or less made up).